pyomo/common/config.py

```diff
@@ -11,6 +11,7 @@
 import os
 import platform
 
+import enum
```

Comment:
FYI: This places a hard dependency on `enum34` for Python 2.7.  I am not entirely sure how to resolve this.

Reply:
This is an unfamiliar problem for me. The issue is that `enum` was added in Python 3.4 and backported to Python 2.7, but there is no guarantee that a user on 2.7 will have the backported enum module? 

My first thought would be to put the import and class definition in a try-except, then mark the test with skipIf.

Reply:
The problem is that this isn't in a test: you need `enum` in the *definition* of the `ConfigEnum` class, so the entire `ConfigEnum` class wouldn't be available if enum wasn't present (only an issue for Python 2.7 at this point).  I think the right answer is to add enum34 to the dependencies for python2.7 in setup.py, but adding hard dependencies is something that we should at least make sure more developer weigh in on.

I will update this PR to include the new dependency and then wait for others to weigh in.

Reply:
@jsiirola It helps that you are a main proponent of keeping 2.7 support in the first place. Fwiw, I believe it makes sense to keep support as long as it doesn't make life too difficult. Most people are content if you don't mess with 3.x.